cmd/chartsvc/handler_test.go

```diff
@@ -0,0 +1,666 @@
+			assert.Equal(t, cv.Values, "/v1/assets/"+tt.chart.ID+"/versions/"+tt.chart.ChartVersions[0].Version+"/values.yaml", "values.yaml resource path should be the same")
+		})
+	}
+}
+
+func Test_newChartResponse(t *testing.T) {
+	tests := []struct {
+		name  string
+		chart models.Chart
+	}{
+		{"chart has only one version", models.Chart{
+			ID: "my-repo/my-chart", ChartVersions: []models.ChartVersion{{Version: "0.1.2"}}},
+		},
+		{"chart has many versions", models.Chart{
+			ID: "my-repo/my-chart", ChartVersions: []models.ChartVersion{{Version: "0.1.2"}, {Version: "0.1.0"}},
+		}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cResponse := newChartResponse(&tt.chart)
+			assert.Equal(t, cResponse.Type, "chart", "response type is chart")
+			assert.Equal(t, cResponse.ID, tt.chart.ID, "chart ID should be the same")
+			assert.Equal(t, cResponse.Attributes.(models.Chart).ChartVersions, tt.chart.ChartVersions, "chart version in the response should be the same")
+		})
+	}
+}
+
+func Test_newChartListResponse(t *testing.T) {
+	tests := []struct {
+		name   string
+		charts []*models.Chart
+	}{
+		{"no charts", []*models.Chart{}},
+		{"has one chart", []*models.Chart{
+			{ID: "my-repo/my-chart", ChartVersions: []models.ChartVersion{{Version: "0.0.1"}}},
+		}},
+		{"has two charts", []*models.Chart{
+			{ID: "my-repo/my-chart", ChartVersions: []models.ChartVersion{{Version: "0.0.1"}}},
+			{ID: "stable/wordpress", ChartVersions: []models.ChartVersion{{Version: "1.2.3"}, {Version: "1.2.2"}}},
+		}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			clResponse := newChartListResponse(tt.charts)
+			assert.Equal(t, len(clResponse), len(tt.charts), "number of charts in response should be the same")
+			for i := range tt.charts {
+				assert.Equal(t, clResponse[i].Type, "chart", "response type is chart")
+				assert.Equal(t, clResponse[i].ID, tt.charts[i].ID, "chart ID should be the same")
+				assert.Equal(t, clResponse[i].Attributes.(models.Chart).ChartVersions, tt.charts[i].ChartVersions, "chart version in the response should be the same")
+			}
+		})
+	}
+}
+
+func Test_newChartVersionResponse(t *testing.T) {
+	tests := []struct {
+		name  string
+		chart models.Chart
+	}{
+		{"my-chart", models.Chart{
+			ID: "my-repo/my-chart", ChartVersions: []models.ChartVersion{{Version: "0.1.0"}, {Version: "0.2.3"}},
+		}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			for i := range tt.chart.ChartVersions {
+				cvResponse := newChartVersionResponse(&tt.chart, tt.chart.ChartVersions[i])
+				assert.Equal(t, cvResponse.Type, "chartVersion", "response type is chartVersion")
+				assert.Equal(t, cvResponse.ID, tt.chart.ID+"-"+tt.chart.ChartVersions[i].Version, "reponse id should have chart version suffix")
+				assert.Equal(t, cvResponse.Attributes.(models.ChartVersion).Version, tt.chart.ChartVersions[i].Version, "chart version in the response should be the same")
```

Comment:
check the selfLink and "chart" relationship here

Reply:
done